src/Features/Core/Diagnostics/AbstractHostDiagnosticUpdateSource.cs

```diff
@@ -44,14 +44,23 @@ protected void RaiseDiagnosticsUpdated(DiagnosticsUpdatedArgs args)
             }
         }
 
-        internal void ReportAnalyzerDiagnostic(DiagnosticAnalyzer analyzer, Diagnostic diagnostic, Workspace workspace, ProjectId projectId)
+        internal void ReportAnalyzerDiagnostic(DiagnosticAnalyzer analyzer, Diagnostic diagnostic, Workspace workspace, ProjectId projectOpt)
         {
             if (workspace != this.Workspace)
             {
                 return;
             }
 
-            var project = workspace.CurrentSolution.GetProject(projectId);
+            // check whether we are reporting project specific diagnostic or workspace wide diagnostic
+            var project = projectOpt != null ? workspace.CurrentSolution.GetProject(projectOpt) : null;
+            
+            // check whether project the diagnostic belong to still exist
+            if (projectOpt != null && project == null)
```

Comment:
Is it necessary to check projectOpt for null twice? If projectOpt is null and project is also null, then it will proceed beyond this if - is that intentional / necessary?